src/main/pages/workspace-admin/installers.adoc

```diff
@@ -29,27 +31,36 @@ Installers are typically language servers and tools that provide features like s
+[id="what-happens-when-enabling-and-disabling-installers"]
+== What happens when enabling and disabling installers 
 
-[id="enable-and-disable-installers"]
-== Enable and Disable Installers
+You can enable or disable installers per machine by using the *User Dashboard* or by updating the workspace machine configuration. When an installer is enabled, the bootstrapper executes an installer script after the workspace has started.
 
-Installers are can be enabled/disabled in User Dashboard per machine, or directly in workspace machine configuration. When an installer is enabled, bootstrapper executes an installer script after the workspace has started.
+The following shows installers that are enabled and disabled:
 
 image::workspaces/installers.png[]
 
-[id="installers-failures"]
-== Installers Failures
+[id="troubleshooting-installer-failures"]
+== Troubleshooting installer failures
+
+Installers run as if a user in the container may have `sudoers` privileges. If the user does not have privileges, the installer fails with `permission denied` issues.
+
+This problem can occur with OpenShift when a pod is run by a user with no permissions to use `sudo` or to access or modify resources on the file system. 
+
+In most cases, this problem can be solved by rebuilding the base image so that it already has all of the dependencies for particular agents that an installer activates. 
 
-As installer scripts may expect that a user in the container/pod is in `sudoers` .e. have privileges, some of them may fail due to `permission denied` issues. This is the case with OpenShift, where a pod is run by an arbitrary user with no permissions to use sudo, and/or access/modify resources on the file system. In most cases, though, this problem can be solved by rebuilding the base image so that it already has all of the dependencies for a particular agents that an installer activates. Another possible issue might be with permissions to files and folders. For example, an installer may need to write to https://github.com/eclipse/che-dockerfiles/blob/master/recipes/stack-base/centos/Dockerfile#L45-L57[user home directory].
+Another possible issue can be with permissions to files and directories. For example, an installer may need to write to the https://github.com/eclipse/che-dockerfiles/blob/master/recipes/stack-base/centos/Dockerfile#L45-L57[user home directory].
 
-== Installer Registry and REST API
+[id="installer-registry-and-rest-api"]
+== Installer registry and REST API
 
-Che installers are stored at Installer Registry. They can be viewed and edited through the Installer Registry REST API:
+Che installers are stored in the Installer Registry. They can be viewed and edited through the Installer Registry REST API:
```

Comment:
The paths (1st column) in the following table should be monospaced, i.e. in backticks.